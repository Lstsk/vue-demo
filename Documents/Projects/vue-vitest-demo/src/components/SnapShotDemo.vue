<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <p>This is a snapshot testing examples.</p>
  </div>
</template>

<script setup>
const props = defineProps({
  msg: {
    type: String,
    default: "Hello, World!",
  },
});
</script>
