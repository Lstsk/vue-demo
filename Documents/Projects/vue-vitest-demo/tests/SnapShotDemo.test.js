import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import SnapShotDemo from "../src/components/SnapShotDemo.vue";

describe("SnapShotDemo.vue Snapshot", () => {
  it("matches the snapshot", () => {
    // Mount the component with a custom prop value
    const wrapper = mount(SnapShotDemo, {
      props: { msg: "Snapshot Test" },
    });

    console.log(wrapper.html());

    // Capture and compare the snapshot of the component's HTML
    expect(wrapper.html()).toMatchSnapshot();
  });
});
