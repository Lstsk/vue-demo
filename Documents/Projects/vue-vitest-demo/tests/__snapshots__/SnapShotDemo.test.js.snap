// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SnapShotDemo.vue Snapshot > matches the snapshot 1`] = `
"<div class="hello">
  <h1>Snapshot Test</h1>
  <p>This is a snapshot testing examples.</p>
</div>"
`;
